lua/basic_command_register.lua

```diff
@@ -0,0 +1,27 @@
+local win = vis.win
+vis:command_register("foo", function(argv, force, win, cursor, range)
+	append(win, '$', 'args: ' .. #argv)
+	for i,arg in ipairs(argv) do
+		append(win, '$', '\t' .. i .. ' ' .. arg)
+	end
+	append(win, '$', 'bang: ' .. (force and "yes" or "no"))
+	append(win, '$', 'name: ' .. win.file.name)
+	append(win, '$', 'pos: ' .. cursor.pos)
+	append(win, '$', 'range: ' .. (range ~= nil and
+				('['..range.start..', '..range.finish..']')
+				or "invalid range"))
+	append(win, '$', '')
+	return true;
+end)
+
+vis:command('foo')
+vis:command('foo!')
+-- TODO Providing a range does not work
```

Comment:
What makes you think it doesn't work? The given range (as all ranges for now) is in byte offsets from the start of the file